### DISCUSSION TITLE
Overcoming Telegram API's 20MB Download Limit for Voice Messages

### DISCUSSION THREAD

**Discussion by tive-tilray** [n3d1117/chatgpt-telegram-bot#440] [Q&A]
Hello Developers,

I'm really excited to be a part of this journey with you.

I'm an amateur coder and a huge enthusiast of the new wave of AI, particularly LLMs. I'm diving deeper into this world to bring as much convenience as possible into everyday life. My teammates and I primarily use Telegram, so we decided to integrate ChatGPT directly into our communication workflow. Everything was going smoothly with my own bot until I hit the 20MB download limit imposed by the Telegram API. This makes the bot somewhat limited for our purposes, as we frequently send voice messages longer than 20 minutes, which tend to be around 20MB in size.

So my question is: Do you have any idea how to overcome this limitation? I've always been a bit apprehensive about contributing to the open-source world, and I've taken a long break from coding. However, the universe is giving me another opportunity to be part of it, and I think this could be a good starting point. Would you consider adding this feature to your code if it's feasible?

My second question is about Whisper's charging model for transcriptions. I assume it's OpenAI-based and uses tokens just like regular text messages? I understand if you'd prefer not to answer this question, as I can probably find the information in OpenAI's documentation (I didn't even know Whisper was from OpenAI).

Oh, and one more thing. Is there any chance to enable transcription for groups somehow? Because it seems like it works only on private chats with bot.

I hope this message finds you well! :)

**Comment by k3it** [comment]
Up to 2000 MB upload is possible with the self hosted Bot API server.   see https://github.com/n3d1117/chatgpt-telegram-bot/pull/504
